src/app/shared-services/db/sqlite.service.ts

```diff
@@ -333,6 +334,74 @@ export class SqliteService {
     )
     return durationDays
   }
+
+  async getStaypoints(id: string): Promise<StayPoints> {
+    await this.ensureDbReady()
+    const { values } = await this.db.query(
+      `SELECT * FROM staypoints WHERE trajectory=? ORDER BY starttime;`,
+      [id]
+    )
+    // empty staypoints are to be expected and are handled in sp service
+    if (!values.length) return undefined
+    const data = values.reduce<StayPoints>(
+      (d, { trajectory, lat, lon, starttime, endtime }) => {
+        d.coordinates.push([lat, lon])
+        d.starttimes.push(convertTimestampToDate(starttime))
+        d.endtimes.push(convertTimestampToDate(endtime))
+        return d
+      },
+      { trajID: id, coordinates: [], starttimes: [], endtimes: [] }
+    )
+    return data
+  }
+
+  async upsertStaypoints(id: string, sp: StayPoints) {
```

Comment:
Again: `trajectoryId`. Also: why not `stayPoints` instead of `sp` to increase readability.